src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Mvc.TagHelpers.Internal;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.WebEncoders;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting &lt;img&gt; elements that supports file versioning.
+    /// </summary>
+    /// <remarks>
+    /// The tag helper won't process for cases with just the 'src' attribute.
+    /// </remarks>
+    [TargetElement("img", Attributes = FileVersionAttributeName)]
+    public class ImageTagHelper : TagHelper
+    {
+        private static readonly string Namespace = typeof(ImageTagHelper).Namespace;
+
+        private const string FileVersionAttributeName = "asp-file-version";
+        private const string SrcAttributeName = "src";
+
+        private FileVersionProvider _fileVersionProvider;
+
+        private static readonly ModeAttributes<Mode>[] ModeDetails = new[] {
+            // Regular src with file version
+            ModeAttributes.Create(Mode.FileVersion, new[] { FileVersionAttributeName }),
+        };
+
+        private enum Mode
+        {
+            /// <summary>
+            /// Just adding a file version for the generated urls.
+            /// </summary>
+            FileVersion = 0
+        }
+
+        /// <summary>
+        /// Src of the image.
+        /// </summary>
+        /// <remarks>
+        /// Passed through to the generated HTML in all cases.
+        /// </remarks>
+        [HtmlAttributeName(SrcAttributeName)]
+        public string Src { get; set; }
+
+        /// <summary>
+        /// Value indicating if file version should be appended to the src urls.
+        /// </summary>
+        /// <remarks>
+        /// If <c>true</c> then a query string "v" with the encoded content of the file is added.
+        /// </remarks>
+        [HtmlAttributeName(FileVersionAttributeName)]
+        public bool? FileVersion { get; set; }
```

Comment:
I agree that AppendFileVersion or AddFileVersion would make more sense. This is modeled off the Link and Script tag helpers. If we change the name here then it would need to be changed on both those tag helpers too.